### PR TITLE
PUBDEV-5084: Added metalearner_fold_column arg to Stacked Ensemble

### PR DESCRIPTION
**Pull request by ledell**
- Adds `metalearner_fold_column` to Stacked Ensemble (defaults to `NULL`)
- If user specifies `nfolds` and `metalearner_fold_column`, the user will receive an error
- If `metalearner_fold_column` is specified, then it's also added to the Level One frame.
- Includes runit, pyunit
- Added parameter to Stacked Ensemble user guide

### CODE REVIEW

File: h2o-algos/src/main/java/hex/schemas/StackedEnsembleV99.java [hunk 1]
```diff
@@ -51,15 +51,13 @@
...
-    // TODO: Add metalearner_fold_column
-    // also requires: import water.api.schemas3.FrameV3;
-    /*
+    // For ensemble metalearner cross-validation
     @API(level = API.Level.secondary, direction = API.Direction.INOUT, gridable = true,
             is_member_of_frames = {"training_frame"},
-            is_mutually_exclusive_with = {"ignored_columns", "response_column", "weights_column", "offset_column"},
-            help = "Column with cross-validation fold index assignment per observation.")
+            //is_mutually_exclusive_with = {"ignored_columns", "response_column", "weights_column", "offset_column"},
+            is_mutually_exclusive_with = {"response_column"},
```
Comment (ledell) [inline h2o-algos/src/main/java/hex/schemas/StackedEnsembleV99.java]: I should check about adding `ignored_columns` back in for now... Stacked Ensemble doesn't have weights or offset support yet, so we don't need those.

File: h2o-algos/src/main/java/hex/StackedEnsembleModel.java [hunk 2]
```diff
@@ -135,7 +133,14 @@ protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_ke
...
     }
 
+    // Add response column to level one frame
     levelOneFrame.add(this.responseColumn, adaptFrm.vec(this.responseColumn));
+    /*
+    // Add metalearner_fold_column to level one frame if it exists
+    if (this._parms._metalearner_fold_column != null) {
+      levelOneFrame.add(this._parms._metalearner_fold_column, adaptFrm.vec(this._parms._metalearner_fold_column));
+    }
+    */
```
Comment (navdeep-G) [inline h2o-algos/src/main/java/hex/StackedEnsembleModel.java]: You can remove the above comment block.

**Comment by navdeep-G** [comment]
Just a minor comment. Rest looks good.

File: h2o-algos/src/main/java/hex/schemas/StackedEnsembleV99.java (hunk 1 above)
Comment (ledell) [reply h2o-algos/src/main/java/hex/schemas/StackedEnsembleV99.java]: Added `ignored_columns` back in.

File: h2o-algos/src/main/java/hex/StackedEnsembleModel.java (hunk 2 above)
Comment (ledell) [reply h2o-algos/src/main/java/hex/StackedEnsembleModel.java]: Thanks, removed.

**Pull request merged**